app/models/friendship.rb

```diff
@@ -0,0 +1,6 @@
+# frozen_string_literal: true
+
+class Friendship < ApplicationRecord
+  belongs_to :user
```

Comment:
- [ ] sender sent a friend request to receiver:
 - sender can't send another request
 - reciever can't send a friend request to sender.

We can achieve this by adding interchangeable validation to check if the [user_id, friend_id] or [friend_id, user_id] exist in the friendship table. [please take a look at this example](https://stackoverflow.com/questions/29123620/rails-validate-unique-friendship-relationships)